test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_session]: block_session.c
+ * The block manager extent list consists of both extent blocks and size blocks. This unit test
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+
+void
+validate_and_cleanup_ext_block(WT_EXT *ext)
```

Comment:
I would move these validation / cleanup functions as close to where they are used.

Reply:
I have organised the ext functions and size functions together along with the validation, cleanup functions. Let me know if it worked better.